src/util/ajax.js

```diff
@@ -129,7 +132,9 @@ function makeFetchRequest(requestParameters: RequestParameters, callback: Respon
 
         const requestTime = Date.now();
 
+        _inflightRequestCount++;
         window.fetch(request).then(response => {
+            _inflightRequestCount = Math.max(_inflightRequestCount - 1, 0);
```

Comment:
Is the `max` there to handle double-decrements in case both an error handler and .cancel() are called? Would doing something id-based be more reliable?

```
const reqId = nextId++;
activeRequests[reqId] = true;
delete activeRequests[reqId];
infLightRequests = Object.keys(activeRequests).length;
```

Reply:
yes, that is correct! I thought about that too, but I was trying to keep the instrumentation as low runtime overhead as possible